src/Model/Behavior/SearchBehavior.php

```diff
@@ -45,8 +45,12 @@ class SearchBehavior extends Behavior
      */
     public function findSearch(Query $query, array $options)
     {
-        if (isset($options['search'])) {
-            $options = $options['search'];
+        if (!isset($options['_search']) && (!isset($options['_filter']) || $options['_filter'] === true)) {
+            $options = $this->filterParams($options);
```

Comment:
The check for `_search` and `_filter` should be mutually exclusive. `filterParams()` should be called after options are extracted from `$options['_search']` below.